lib/pubid_iso/urn.rb

```diff
@@ -0,0 +1,49 @@
+module PubidIso
+  class URN
+    attr_accessor :identifier
+
+    def initialize(identifier)
+      @identifier = identifier
+    end
+
+    def to_s
+      result = "urn:iso:std:#{publisher}:#{identifier.number}"
+
+      if identifier.part
+        result += ":-#{identifier.part}"
+      end
+
+      if identifier.stage
+        result += ":stage-#{stage}"
+      end
+      result
+    end
+
+    def stage
```

Comment:
The URN needs to support all stages and substages. (https://www.iso.org/stage-codes.html)

Reply:
Done.